### DISCUSSION TITLE
OrientDB freezing all of sudden every 20-30 days unfrequently

### DISCUSSION THREAD

**Discussion by arnabhik** [orientechnologies/orientdb#9566] [Q&A]
Hi Guys,

We are using OrientDB 3.1.3 on Distributed Mode on 3 Cluster with 16Core/64GB 

We have around 1 Million Nodes and 5 Million Edges.

We are using 2GB-4GB Heap Size

Unfortunately 50% of our data is updating daily, so we are using batch.

Also we are doing around 40-50 Queries per sec on OrientDB on API (select queries with joins using in() or out() )

The problem is every 20-30 days orientdb is freezing and no one can connect either from API or Console.

We have to run jcmd <> GC.run and it starts again,

It happens to one node at a time.

We are not able to figure out the problem can some one help 

Similarly the on Inserts the Memory goes super high, even on update.

The size of the data is no more then 100MB but orient consumes more then 8 GB.

**Comment by luigidellaquila** [comment]
Hi @arnabhik 

OrientDB v 3.1.3 has some known issues with the distributed clustering, there is a significant chance that this problem is already fixed, so my suggestion is to upgrade to latest version (3.1.11 now)

Thanks

Luigi

**Reply by arnabhik** [reply]
Sure,

Will do then.


Also i saw clearing the Heap Fixes this, but sometimes even heap clearing using jcmd <> GC.run stucks.


And secondly,

OrientDB is using too much memory for a small 1 million dataset.

Is it because of deletes and inserts that are run every day.


How can i clear the ram, any way to flush from python
